wrap/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+cmake_minimum_required(VERSION 3.9)
+
+# Set the project name and version
+project(GTwrap VERSION 1.0)
+
+# ##############################################################################
+# General configuration
+
+set(WRAP_PYTHON_VERSION
+    "Default"
+    CACHE STRING "The Python version to use for wrapping")
+
+include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/GtwrapUtils.cmake)
+gtwrap_get_python_version(${WRAP_PYTHON_VERSION})
+
+# ##############################################################################
+# Install the CMake file to be used by other projects
+if(WIN32 AND NOT CYGWIN)
+  set(SCRIPT_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/CMake")
+else()
+  set(SCRIPT_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib/cmake")
+endif()
+
+# Install scripts to the standard CMake script directory.
+install(FILES cmake/gtwrapConfig.cmake cmake/PybindWrap.cmake
+              cmake/GtwrapUtils.cmake
+        DESTINATION "${SCRIPT_INSTALL_DIR}/gtwrap")
+
+# Install wrapping scripts as binaries to `CMAKE_INSTALL_PREFIX/bin` so they can
+# be invoked for wrapping.
+install(PROGRAMS scripts/pybind_wrap.py scripts/matlab_wrap.py TYPE BIN)
+
+# Install pybind11 directory to `CMAKE_INSTALL_PREFIX/lib/pybind11` This will
+# allow the gtwrapConfig.cmake file to load it later.
+install(DIRECTORY pybind11 TYPE LIB)
+
+# ##############################################################################
+# Install the Python package
+find_package(
+  Python ${WRAP_PYTHON_VERSION}
+  COMPONENTS Interpreter
+  REQUIRED)
+
+# Detect virtualenv and set Pip args accordingly
+# https://www.scivision.dev/cmake-install-python-package/
+if(DEFINED ENV{VIRTUAL_ENV} OR DEFINED ENV{CONDA_PREFIX})
```

Comment:
Why is this in CMakeLists? TBH nothing in CMakeLists should ever make changes to the system. Anything that modifies the environment should be at install time (`make install`).

Reply:
Unfortunately there is no way to around this because the wrapping is done during `make`, and the gtwrap package is required to be installed at that point.

This is one of those instances where we can't quite follow the standard conventions to achieve the end result.

Reply:
I still think this is probably not what @dellaert wants, but nvm if it is :)

Also, I don't really see the reason `gtwrap` needs to be installed to be used, we can just keep it in the tree.

Reply:
If we don't install it, then we get all sorts of import errors in Python. This is a major failing of Python that imports and relative imports don't work as specified.

Reply:
> I still think this is probably not what @dellaert wants, but nvm if it is :)

I don't understand this. The idea is that we set up the install of gtwrap as a prerequisite so that at make time, the system has everything it needs to finish successfully. 🙂

Reply:
Oh, I got what you mean, but I think if we need to install `gtwrap`, then it should be self-contained (a.k.a. installed to a fresh venv in the build folder) and not touch the current user environment.

Reply:
Yeah that seemed like more work than necessary TBH. It's just a small package that doesn't have any collisions or messes with the standard library so it seemed fine to me, especially as this vastly simplifies the process on the user's end.